features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java

```diff
@@ -82,4 +89,11 @@ public boolean getNewSuspectOnTrap() {
 		return m_snmpV3Users;
 	}
 
+	@Override
+	public void onUpdate(TrapdConfiguration config) {
+		this.m_snmpTrapAddress = config.getSnmpTrapAddress();
+		this.m_snmpTrapPort =config.getSnmpTrapPort();
+		this.m_snmpV3Users=new ArrayList<SnmpV3User>(TrapReceiverImpl.addToSnmpV3Users(config).values());
```

Comment:
Actually as Part of reusability we have added Synchronized block inside TrapReceiverImpl.addToSnmpV3Users and same we are calling here.I hope this serves the cause or If still synchronized is needed even for this block i can update it.Please let me know if its necessary to wrap here as well.